Vector.c

```diff
@@ -290,16 +283,22 @@ void Vector_compact(Vector* this) {
 
    int idx = this->dirty_index;
 
-   /* one deletion: use memmove, which should be faster */
+   // one deletion: use memmove, which should be faster
    if (this->dirty_count == 1) {
       memmove(&this->array[idx], &this->array[idx + 1], (this->items - idx - 1) * sizeof(this->array[0]));
+      this->array[this->items - 1] = NULL;
    } else {
-      /* multiple deletions */
+      // multiple deletions
       for (int i = idx + 1; i < size; i++) {
          if (this->array[i]) {
             this->array[idx++] = this->array[i];
          }
       }
+      // idx is now at the end of the vector and on the first index which should be set to NULL
+      for (int i = idx; i < size; i++)
+      {
```

Comment:
Should be one line too …
```suggestion
      for (int i = idx; i < size; i++) {
```

Reply:
or even `memset(&this->array[idx], '\0', (size - idx) * sizeof(this->array[0]));`?